apps/search/js/providers/suggestions.js

```diff
@@ -37,13 +37,15 @@
     },
 
     search: function(input) {
-      return new Promise((resolve, reject) => {
-        var url = encodeTerms(SearchProvider('suggestUrl'), input);
-        LazyLoader.getJSON(url, true).then(result => {
-          var results = result[1];
-          // We add an item to search the entered term as well
-          results.unshift(result[0]);
-          resolve(results);
+      return SearchProvider.ready().then(resolve => {
```

Comment:
Ok, I think I mis-read the code before, sorry.  You're right, I think then() should just return a new promise as you're doing here.

This looks good, I think you should remove the resolve here then, that was wrong.

```js
return SearchProvider.ready().then(() => {
```